mm2src/coins/eth.rs

```diff
@@ -3185,6 +3187,44 @@ impl<T: TryToAddress> TryToAddress for Option<T> {
     }
 }
 
+pub trait GuiAuthMessages {
+    fn gui_auth_sign_message_hash(message: String) -> Option<[u8; 32]>;
+    fn generate_gui_auth_signed_validation(generator: GuiAuthValidationGenerator)
+        -> SignatureResult<serde_json::Value>;
+}
+
+impl GuiAuthMessages for EthCoin {
+    fn gui_auth_sign_message_hash(message: String) -> Option<[u8; 32]> {
+        let message_prefix = "atomicDEX Auth Ethereum Signed Message:\n";
+        let prefix_len = CompactInteger::from(message_prefix.len());
+
+        let mut stream = Stream::new();
+        prefix_len.serialize(&mut stream);
+        stream.append_slice(message_prefix.as_bytes());
+        stream.append_slice(message.len().to_string().as_bytes());
+        stream.append_slice(message.as_bytes());
+
+        Some(keccak256(&stream.out()).take())
+    }
+
+    fn generate_gui_auth_signed_validation(
+        generator: GuiAuthValidationGenerator,
+    ) -> SignatureResult<serde_json::Value> {
```

Comment:
Should we return a specific struct here instead of `serde_json::Value`? I guess it might be also shared between API and auth codebases.